webapp/channels/src/components/admin_console/data_grid/data_grid_row.tsx

```diff
@@ -9,48 +9,57 @@ import type {Row, Column} from './data_grid';
 
 import './data_grid.scss';
 
-type Props = {
+type DataGridRowProps = {
     columns: Column[];
     row: Row;
 }
 
-class DataGridRow extends React.Component<Props> {
-    renderCell(row: Row, column: Column) {
-        const style: CSSProperties = {};
-        if (column.width) {
-            style.flexGrow = column.width;
-        }
-
-        if (column.textAlign) {
-            style.textAlign = column.textAlign;
-        }
-
-        if (column.overflow) {
-            style.overflow = column.overflow;
-        }
-
-        return (
-            <div
-                key={column.field}
-                className={classNames('DataGrid_cell', column.className)}
-                style={style}
-            >
-                {row.cells[column.field]}
-            </div>
-        );
+type DataGridCellProps = {
+    column: Column;
+    row: Row;
+}
+
+const DataGridCell = ({row, column}: DataGridCellProps) => {
+    const style: CSSProperties = {};
+    if (column.width) {
+        style.flexGrow = column.width;
     }
 
-    render() {
-        const cells = this.props.columns.map((col) => this.renderCell(this.props.row, col));
-        return (
-            <div
-                className='DataGrid_row'
-                onClick={this.props.row.onClick}
-            >
-                {cells}
-            </div>
-        );
+    if (column.textAlign) {
+        style.textAlign = column.textAlign;
     }
-}
 
-export default DataGridRow;
+    if (column.overflow) {
+        style.overflow = column.overflow;
+    }
+
+    return (
+        <div
+            key={column.field}
+            className={classNames('DataGrid_cell', column.className)}
+            style={style}
+        >
+            {row.cells[column.field]}
+        </div>
+    );
+};
+
+const DataGridRow = ({row, columns}: DataGridRowProps) => {
+    const cells = columns.map((column, index) => (
+        <DataGridCell
+            key={index}
```

Comment:
In general, using the index as a key is not a good idea (https://react.dev/learn/rendering-lists#why-does-react-need-keys).

Nevertheless, at this point, I see no better way to handle this (the key should be the field name, but I fear is not that unique or something we can use here), after first render these shouldn't change (there is no reorder of columns nor inserts/deletes), and in theory it is what React was doing either way under the hood, so let's keep it like this.

Reply:
Thanks for that info! I did see the warning, but did not have a good idea of what else to use and saw that `key={index}` has been used in other place.